pkg/services/generic.go

```diff
@@ -333,3 +344,66 @@ func (s *sqlGenericService) treeWalkForSqlizer(listCtx *listContext, tslTree tsl
 
 	return tslTree, sqlizer, nil
 }
+
+func isJSONBSearch(search string) bool {
+	if strings.Contains(search, "->>") || strings.Contains(search, "@>") {
+		return true
+	}
+	return false
+}
+
+func validateJSONBSearch(search string) error {
+	// handle ->> search. for example: payload -> 'data' -> 'manifest' -> 'metadata' -> 'labels' ->> 'foo' = 'bar'
+	if strings.Contains(search, "->>") {
+		jsonbStr := stringSplitwithTrim(search, "->>")
+		if err := isValidFields(stringSplitwithTrim(jsonbStr[0], "->")); err != nil {
+			return fmt.Errorf("the search field name is invalid: %v", err)
+		}
+		labelStr := stringSplitwithTrim(jsonbStr[1], "=")
+		if err := validation.IsQualifiedName(labelStr[0]); err != nil {
+			return fmt.Errorf("the search name is invalid: %v", err)
+		}
+		if len(labelStr) > 1 {
+			if err := validation.IsValidLabelValue(labelStr[1]); err != nil {
+				return fmt.Errorf("the search value is invalid: %v", err)
+			}
+		}
+	}
+	// handle @> search. for example: payload -> 'data' -> 'manifests' @> '[{\"metadata\":{\"labels\":{\"foo\":\"bar\"}}}]'
+	if strings.Contains(search, "@>") {
+		jsonbStr := stringSplitwithTrim(search, "@>")
+		if err := isValidFields(stringSplitwithTrim(jsonbStr[0], "->")); err != nil {
+			return fmt.Errorf("the search field name is invalid: %v", err)
+		}
+		// validate the value is a valid json object
+		if !isValidJSONObject(jsonbStr[1]) {
```

Comment:
Do we support any json string?

Reply:
Yes. support any json only if it is valid json object.

Reply:
do we need to support any json string? For instance, even a spec field in the json string?